valhalla/filesystem.h

```diff
@@ -187,16 +189,47 @@ class directory_entry {
   filesystem::path path_;
 };
 
+class directory_iterator {
```

Comment:
to do recursive delete, we need a non-recursive directory iterator, go figure. the recursion happens inside of the delete not at the iterator level